### PR TITLE
 Enable DeepseekV2 Lite/Chat models

### PR DESCRIPTION
**Pull request by hlin99**
1, reshape posotions/hidden_state according to hpu requirement
2, w/a of SW-208144

### CODE REVIEW

File: vllm/model_executor/models/deepseek_v2.py [hunk 1]
```diff
@@ -290,7 +310,7 @@ def forward(
         kv_a, _ = latent_cache.split(
             [self.kv_lora_rank, self.qk_rope_head_dim], dim=-1)
         latent_cache = latent_cache.unsqueeze(1)
-        kv_a = self.kv_a_layernorm(kv_a.contiguous())
+        kv_a = self.kv_a_layernorm(kv_a.contiguous().unsqueeze(0)).squeeze(0) # w/a of SW-208144
```
Comment (michalkuligowski) [inline vllm/model_executor/models/deepseek_v2.py:313]: This should be under if_hpu

File: vllm/model_executor/models/deepseek_v2.py [hunk 2]
```diff
@@ -276,9 +290,15 @@ def forward(
...
+        if is_hpu:
+            # need reshape from tensor(x0, y0) to tensor(x1) for hpu
+            _batch_size = positions.shape[0]
+            positions = positions.reshape(positions.shape[0] * positions.shape[1])
+            hidden_states = hidden_states.reshape(hidden_states.shape[0] * hidden_states.shape[1], hidden_states.shape[2])
         if self.q_lora_rank is not None:
-            q = self.q_a_proj(hidden_states)[0]
-            q = self.q_a_layernorm(q)
+            # w/a of SW-208144
+            q = self.q_a_proj(hidden_states)[0].unsqueeze(0)
```
Comment (michalkuligowski) [inline vllm/model_executor/models/deepseek_v2.py:300]: This should be under if_hpu

**Review by michalkuligowski — CHANGES_REQUESTED** [review]

File: vllm/model_executor/models/deepseek_v2.py (hunk 1 above)
Comment (hlin99) [reply vllm/model_executor/models/deepseek_v2.py]: will open a new PR

File: vllm/model_executor/models/deepseek_v2.py (hunk 2 above)
Comment (hlin99) [reply vllm/model_executor/models/deepseek_v2.py]: https://github.com/HabanaAI/vllm-fork/pull/516

File: vllm/model_executor/models/deepseek_v2.py (hunk 1 above)
Comment (hlin99) [reply vllm/model_executor/models/deepseek_v2.py]: https://github.com/HabanaAI/vllm-fork/pull/516

**Pull request closed without merging**